src/pytorch_lightning/strategies/ipu.py

```diff
@@ -142,8 +120,7 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self._optimizer_zero_grad_original = self.lightning_module.optimizer_zero_grad
         self._disable_zero_grad()
 
-        model = LightningIPUModule(self.lightning_module, self.precision_plugin.precision)
-        self.model = model
+        self.model = _LightningModuleWrapperBase(self.lightning_module)
```

Comment:
IPUs can only trace the `forward` method? not the `training_step/validation_step`, etc...?

Reply:
The `forward` method of the `model` argument passed to `poptorch.{training/inference}Model` is currently traced. Before this PR, the `LightningIPUModule` was being passed as the `model` argument, so its `forward` was being traced. This was why the cast was happening on the IPU instead of the host.